### PR TITLE
Make the lesson TOC collapsable on mobile

### LINKED ISSUE #3092: Redesign the lesson TOC mobile view to make it collapsable
It should look like this on mobile:

image
image

Inspiration from https://auto-animate.formkit.com/

### PR DESCRIPTION
**Pull request by flacial**
### Changes

- Instead of the TOC taking the full space on mobile, it's now closed by default
- Update the `subLesson` test file
- Update the `subLesson` style file

### UI screenshots

Always opened
 image
Closed 
image
Opened
 image

### Related PRs

- Closes #3092

### CODE REVIEW

File: pages/curriculum/[lessonSlug]/[subLessonSlug].tsx [hunk 1]
```diff
@@ -26,25 +26,45 @@ import ScrollTopArrow from '../../../components/ScrollTopArrow'
...
+  breakpoint: boolean
+  setToggle: React.Dispatch<React.SetStateAction<boolean>>
 }
-const SubLessonPage: React.FC<Props> & WithLayout = ({
-  selectedSubLessonIndex,
-  lessonSlug,
-  subLessonSlug,
-  subLessons,
-  githubFilePath
+const CustomToggle: React.FC<PropsWithChildren<CustomToggleProps>> = ({
```
Comment (flacial) [inline pages/curriculum/[lessonSlug]/[subLessonSlug].tsx:36]: The only reason I created a custom toggle is to control when the accordion is closed/opened and prevent it from getting opened on desktop view.

File: pages/curriculum/[lessonSlug]/[subLessonSlug].tsx [hunk 2]
```diff
@@ -72,17 +144,10 @@ const SubLessonPage: React.FC<Props> & WithLayout = ({
...
-        style={{
-          position: breakpoint ? 'sticky' : 'initial'
-        }}
-      >
-        <ul className={`p-md-4 ${styles.toc__container} m-0 rounded`}>
-          {mapHeadingsToLi}
-        </ul>
-      </div>
+      <TableOfContents
+        selectedSubLesson={selectedSubLesson as SubLesson}
```
Comment (scotthallock) [inline pages/curriculum/[lessonSlug]/[subLessonSlug].tsx:148]: Is type assertion required here too, in addtion to line 136?

**Review by scotthallock — APPROVED** [review]
LGTM

File: pages/curriculum/[lessonSlug]/[subLessonSlug].tsx (hunk 2 above)
Comment (flacial) [reply pages/curriculum/[lessonSlug]/[subLessonSlug].tsx:148]: Yes. Without it, it would complain that the types don't match.

Comment (flacial) [reply pages/curriculum/[lessonSlug]/[subLessonSlug].tsx:148]: After removing the type assertion, it doesn't complain. Nice catch, I'll update the code!

File: scss/subLessonSlug.module.scss [hunk 3]
```diff
@@ -1,19 +1,32 @@
...
-  width: fit-content;
+  background: colors.$fg-light;
   height: fit-content;
 }
+
+.toc__header {
+  background-color: colors.$bg-white;
+}
+
+.toc__toggle {
```
Comment (SlyBouhafs) [inline scss/subLessonSlug.module.scss:25]: The toggle should probably have a hover effect or something to signal that the element is clickable.

Comment (SlyBouhafs) [reply scss/subLessonSlug.module.scss:25]: Nevermind.

Comment (flacial) [reply scss/subLessonSlug.module.scss:25]: Yeah, it's not clickable on the desktop as it's always open.

**Pull request merged**